core/app.go

```diff
@@ -102,6 +103,15 @@ func NewApp(configFlag string) (*App, error) {
 	a.Tasks = cfg.Tasks
 	a.Telemetry = cfg.Telemetry
 	a.ConfigFlag = configFlag
+
+	// set an environment variable for each service IP address so that
+	// forked processes have access to this information
+	for _, service := range a.Services {
+		envKey := fmt.Sprintf("CONTAINERPILOT_%v_IP",
+			strings.ToUpper(service.Name))
```

Comment:
> I think we should adopt the same convention. service names should be alpha-numeric with dashes. 

Looking at it again, this suggests to me that we should do this in the `config` package and validate the service names there (I'd say do it in the `services` package but we need to do it for `backends` as well). I'm against silently converting/canonicalizing and think we should return an error if the name is bad. This will cause ContainerPilot to halt on initial start, but I'd be willing to discuss it if people have objections.

i.e. the behavior I'm proposing is:
- `my-service` => `my-service` service name and `CONTAINERPILOT_MY_SERVICE_IP` env var
- `my.service` => return error
- `my%service` => return error